This patch file was created on Tue May 12 12:53:38 MST 1992.

This patch is for an RS/6000 port of xlock (sccs version 23.21)
patchlevel = 2.3. This patch depends on "AIXV3" being defined. The
environment was:

  IBM RS/6000 550, AIX 3.1.6, bsdcc (as defined by /usr/lpp/bos/bsdport),
  X11R5pl10.

    Cheers,

    Tom McConnell

 Tom McConnell          |     Internet: tmcconne@sedona.intel.com
 Intel, Corp. C3-21     |     Phone: (602)-554-8229
 5000 W. Chandler Blvd. | The opinions expressed are my own. No one in 
 Chandler, AZ  85226    | their right mind would claim them.

*** ./rotor.c_orig	Tue May 12 11:54:11 1992
--- ./rotor.c	Tue May 12 12:49:11 1992
***************
*** 31,35 ****
  
  #define IDENT(X)	X
! #ifdef __STDC__
  #define CAT(X,Y)	X##Y
  #else
--- 31,35 ----
  
  #define IDENT(X)	X
! #if defined (__STDC__) || defined (AIXV3)
  #define CAT(X,Y)	X##Y
  #else
*** ./resource.c_orig	Tue May 12 11:55:57 1992
--- ./resource.c	Tue May 12 11:56:55 1992
***************
*** 105,109 ****
--- 105,113 ----
  #define DEF_DISPLAY	":0"
  #define DEF_MODE	"life"
+ #ifndef AIXV3
  #define DEF_FONT	"-b&h-lucida-medium-r-normal-sans-24-*-*-*-*-*-iso8859-1"
+ #else AIXV3
+ #define DEF_FONT	"fixed"
+ #endif AIXV3
  #define DEF_BG		"White"
  #define DEF_FG		"Black"
*** ./xlock.h_orig	Tue May 12 11:57:02 1992
--- ./xlock.h	Tue May 12 12:40:01 1992
***************
*** 64,75 ****
  
  #include <sys/time.h>
  #include <poll.h>
  #include <shadow.h>
  #define srandom srand
  #define random rand
  #define MAXRAND (32767.0)
! #define passwd spwd
! #define pw_passwd sp_pwdp
! #define getpwnam getspnam
  
  #else
--- 64,92 ----
  
  #include <sys/time.h>
+ #define passwd spwd
+ #define pw_passwd sp_pwdp
+ #ifdef AIXV3
+ #undef NULL
+ #define NULL 0
+ #include <sys/poll.h>
+ 
+ struct passwd {
+         char    *pw_name;
+         char    *pw_passwd;
+         uid_t   pw_uid;
+         gid_t   pw_gid;
+         char    *pw_gecos;
+         char    *pw_dir;
+         char    *pw_shell;
+ };
+ #else AIXV3
  #include <poll.h>
  #include <shadow.h>
+ #define getpwnam getspnam
+ #endif AIXV3
  #define srandom srand
  #define random rand
  #define MAXRAND (32767.0)
! 
  
  #else
*** ./xlock.c_orig	Tue May 12 12:15:55 1992
--- ./xlock.c	Tue May 12 12:50:52 1992
***************
*** 417,424 ****
      struct passwd *pw;
  
!     pw = getpwnam("root");
      strcpy(rootpass, pw->pw_passwd);
  
!     pw = getpwnam(cuserid(NULL));
      strcpy(userpass, pw->pw_passwd);
  
--- 417,424 ----
      struct passwd *pw;
  
!     pw = (struct passwd *)getpwnam("root");
      strcpy(rootpass, pw->pw_passwd);
  
!     pw = (struct passwd *)getpwnam(cuserid(NULL));
      strcpy(userpass, pw->pw_passwd);
  
